### PR TITLE
feat: add lint, typecheck and test validations to pr's

### PR DESCRIPTION
**Pull request by Jorg88**

### CODE REVIEW

File: .github/workflows/validate-pr.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          indicators: true
+          output: both
+          thresholds: '79 99'
+
+      - name: Find PR number
+        uses: jwalton/gh-find-current-pr@v1
+        id: findPr
+        if: github.ref_name != 'main'
+        with:
+          state: open
```
Comment (tijmenstor) [inline .github/workflows/validate-pr.yaml:111]: This should not be necessary anymore, the PR number is part of the context variables

File: .github/workflows/validate-pr.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        if: steps.cache-dependencies.outputs.cache-hit != 'true'
+        run: pipenv install -d --deploy
+
+      - name: Check types
+        run: pipenv run check-types
+
+      - name: Check types test
+        run: pipenv run check-types-test
+
+  Test:
```
Comment (tijmenstor) [inline .github/workflows/validate-pr.yaml:64]:
```suggestion
  test:
```

File: .github/workflows/validate-pr.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Check formatting
+        run: pipenv run format
+
+      - name: Run lint
+        run: pipenv run lint
+
+      - name: Run test code lint
+        run: pipenv run lint-test
+
+  TypeCheck:
```
Comment (tijmenstor) [inline .github/workflows/validate-pr.yaml:38]:
```suggestion
  typecheck:
```

File: .github/workflows/validate-pr.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+jobs:
+  Lint:
+    name: 🧽 Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        id: cache-dependencies
+        with:
+          python-version: '3.11'
```
Comment (tijmenstor) [inline .github/workflows/validate-pr.yaml:17]: Is it compatible with 3.12 already? Or is that a whole different beast to tackle.

Comment (Jorg88) [reply .github/workflows/validate-pr.yaml:17]: I don't know really, I think updating is a task by itself. There also is a 3.13 already, and we have quite some outdated package versions in the Pipfile. I'll add it to my list of todo's for gh-mpyl

File: .github/workflows/validate-pr.yaml [hunk 5]
```diff
@@ -0,0 +1,110 @@
+---
+
+name: Validate pull requests
+
+on:
+  pull_request:
+
+jobs:
+  lint:
+    name: 🧽 Lint
```
Comment (ptab) [inline .github/workflows/validate-pr.yaml:10]: This emoji is great! 😂

Comment (Jorg88) [reply .github/workflows/validate-pr.yaml:10]: All credits to Sam for that! Hehe

**Pull request merged**